src/extractors/paths.py

```diff
@@ -0,0 +1,16 @@
+from algorithms.uast_to_bag_paths import Uast2BagOfPaths
+from sourced.ml.extractors import BagsExtractor, register_extractor
+from sourced.ml.algorithms import NoopTokenParser
+
+
+@register_extractor
+class BagOfPathsExtractor(BagsExtractor):
+    NAME = "id"
+    NAMESPACE = "i."
```

Comment:
it is a namespace for identifiers, better to take another one.

Reply:
not sure what the namespace is and how it's used (just copied sourced.ml example), could you clarify?

Reply:
The base `BagsExtractor` class is parent of the extractors we use, e.g. the one that extract identifiers. In order to distinguish features of different types , we prepend to each feature this single character  NAMESPACE`, e.g. for identifiers features could look like: `i.var1, i.var2, i.func1`, but for literals : `l.get, l.set`, etc. We do this because we usually don't extract only one type of feature. Anyway, I think you can use `v.` here